sql/sp.cc

```diff
@@ -968,7 +968,22 @@ enum_sp_return_code sp_drop_routine(THD *thd, enum_sp_type type,
         name->m_db.str, name->m_name.str, &routine);
   if (error) return SP_INTERNAL_ERROR;
 
+#ifdef WITH_WSREP
+  if (routine == nullptr) {
+    if (thd->lex->drop_if_exists) {
+      /* If 'IF EXISTS' clause is present, we replicate always.
+         In such a case binlogging part is done on the caller level. */
+      if (WSREP(thd) &&
+          wsrep_to_isolation_begin(thd, WSREP_MYSQL_DB, NULL, NULL)) {
```

Comment:
I know that we us "mysql" db for all SPs during TOI. I wonder why don't we use the exact name of the database?
```suggestion
          wsrep_to_isolation_begin(thd, name->m_db.str, NULL, NULL)) {
```

Reply:
As discussed on Slack, for now, let's leave it consistent across the current code